frame/frame.go

```diff
@@ -0,0 +1,181 @@
+package frame
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"github.com/datastax/go-cassandra-native-protocol/message"
+	"github.com/datastax/go-cassandra-native-protocol/primitive"
+)
+
+// A high-level representation of a frame, where the body is fully decoded.
+type Frame struct {
+	Header *Header
```

Comment:
💯 I like that both `Frame` and `RawHeader` now have the same `Header` type, it didn't make sense how the types were different before.

Reply:
Indeed it's much better this way. I almost regret that we didn't do it in Java :-D . The only thing that bothers me a bit is `Header.BodyLength` – you can't reasonably expect users to guess the right body length when creating a message. So the "contract" is to leave it zero and let the codec set the body length for you. This works fine, the only drawback being that the codec sometimes modifies the input frame when setting the body length on its header.